src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1074,35 +1076,30 @@ static bool processLocation
   {
     const ContextAttribute* caP = caV[ix];
 
-    for (unsigned jx = 0; jx < caP->metadataVector.size(); ++jx)
```

Comment:
Hard to follow in diff mode. I'd recommend to have a look to the final code of this function directly.